### PR TITLE
Docstrings for vectorfields and optimize

### PR DESCRIPTION
**Pull request by Louisbu**

### CODE REVIEW

**Comment by ShrimpyLouis** [comment]
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

File: hybrid_routing/optimize.py [hunk 1]
```diff
@@ -22,7 +22,51 @@ def optimize_route(
...
+
+    This function first computes the locally optimized paths with Scipy's ODE solver. Given the starting coordinates (x_start, y_start), time (t_max), speed of the ship (vel), and the direction the ship points in (angle_amplitude / num_angles), the ODE solver returns a list of points on the locally optimized path.
+
+    We then use a loop to compute all locally optimal paths with given angles in the angle amplitude and store them in a list.
+
+    We next finds the list of paths with an end point (x1, y1) that has the smallest Euclidean distance to the destination (x_end, y_end).
+
+    We then use the end point (x1, y1) on that path to compute the next set of paths by repeating the above algorithm.
+
+    This function terminates till the last end point is within a neighbourhood of the destination (defaults 3 * vel / 4).
```
Comment (daniprec) [inline hybrid_routing/optimize.py]: The lines should respect the character limit, even in comments. That will ease the read of the docstring. Also, there is no need of double linescapes.

File: hybrid_routing/vectorfields/base.py [hunk 2]
```diff
@@ -7,30 +9,79 @@
...
         return duy(x, y)[1]
 
-    def wave(self, p, t, vel=jnp.float16(0.1)):
+    def wave(
+        self,
+        p: Iterable[float, float, float],
+        t: Iterable[float],
+        vel: jnp.float16 = jnp.float16(0.1),
+    ) -> Iterable[float, float, float]:
+        """_summary_
```
Comment (daniprec) [inline hybrid_routing/vectorfields/base.py]: Fill this summary.

**Review by daniprec — CHANGES_REQUESTED** [review]
Very good! I recommend to review some docstrings and cut lines when they exceed the character limit.

File: hybrid_routing/optimize.py (hunk 1 above)
Comment (Louisbu) [reply hybrid_routing/optimize.py]: Thanks! It's fixed now.

File: hybrid_routing/vectorfields/base.py (hunk 2 above)
Comment (Louisbu) [reply hybrid_routing/vectorfields/base.py]: fixed!

**Review by daniprec — APPROVED** [review]
Good to go!

**Pull request merged**